src/types/api/user.ts

```diff
@@ -0,0 +1,25 @@
+import { isDate, parseISO } from 'date-fns'
+import { Opaque } from 'type-fest'
+import { z } from 'zod'
+
+import { Agency } from './agency'
+
+type UserId = Opaque<string, 'UserId'>
+type UserContact = Opaque<string, 'UserContact'>
+
+const DateString = z.string().refine(
+  (val) => isDate(parseISO(val)),
+  (val) => ({ message: `${val} is not a valid date` }),
+)
+
+export const User = z.object({
```

Comment:
I think you can treat this as a class, where the types and the implementation are the same. But since you've raised it up a couple of times I'll change it 😆